.github/workflows/validate-gha-refs.yml

```diff
@@ -0,0 +1,68 @@
+name: Validate GHA References
```

Comment:
I moved the logic into an action, and moved the PowerShell code into it's own .ps1 file to be consistent with how other actions using PowerShell.

I put the logic in it's own workflow based on a comment from @BenedekFarkas to run this check only on pushes to the dev branch.
 >I think this is a good compromise in terms of complexity vs. how much it achieves. I'd rather put this in a separate workflow, because you're not really validating the pull request anymore. Although it's important that whoever merged it (or contributed to the PR) is notified in case of a failure, but that's up to personal settings too.

I temporarily added the `issue/OSOE-517` branch to the push trigger to demonstrate what the output summary would look like. I avoided adding `pull_request` as a trigger for now because so many ref mismatches are found. I think this is still a painful developer experience.

To solve this, I think I can add a step to the [validate-this-gha-refs](https://github.com/Lombiq/GitHub-Actions/blob/issue/OSOE-517/.github/workflows/validate-this-gha-refs.yml) workflow to get a list of changed/added files that are part of the pull request, and only consider those when checking GHA refs. 

This way the workflow only fails on files that are important to check. For example, if a PR is opened to change the README.md file, it is not important to check the refs of all the GHA files because the PR only concerns the readme file.

Before I implement that, though, I want to get your feedback to make sure that is desired. 

I attempted to implement it but it is more work than I expected. But I think it sort of addresses the management issue.

Reply:
> I moved the logic into an action, and moved the PowerShell code into it's own .ps1 file to be consistent with how other actions using PowerShell.

Yes that's nice!

> I put the logic in it's own workflow based on a comment from @BenedekFarkas

Ah, sorry I somehow skipped that. Then okay!

> To solve this, I think I can add a step to the [validate-this-gha-refs](https://github.com/Lombiq/GitHub-Actions/blob/issue/OSOE-517/.github/workflows/validate-this-gha-refs.yml) workflow to get a list of changed/added files that are part of the pull request, and only consider those when checking GHA refs.

How this would work exactly? I guess for `dev` it would be the same => checking everything.

And for a PR, if I would modify let's say the [Login To Azure action](https://github.com/Lombiq/GitHub-Actions/blob/dev/.github/actions/login-to-azure/action.yml) on `issue/xxx-123` branch, then it would check if where that action is used, it's using the `issue/xxx-123` branch, and only warn about those. Is that correct?

Reply:
>How this would work exactly? I guess for dev it would be the same => checking everything.

Yes, for `dev` it's the easier case because we can check everything.

>And for a PR, if I would modify let's say the [Login To Azure action](https://github.com/Lombiq/GitHub-Actions/blob/dev/.github/actions/login-to-azure/action.yml) on issue/xxx-123 branch, then it would check if where that action is used, it's using the issue/xxx-123 branch, and only warn about those. Is that correct?

Yes, that is correct. If the `Login to Azure` action is modified in any way, then any GHA file that calls it with `uses: Lombiq/GitHub-Actions/.github/actions/login-to-azure@...` will need to be checked for using `@issue/xxx-123`. 

I just realized that this will be tricky with actions because any of the files under the action's folder might change, not just the `.yaml` file.

Maybe the straightforward solution is to add a new `Update-GHA-Refs-to-Branch.ps1` script to the [Scripts](https://github.com/Lombiq/GitHub-Actions/tree/dev/Scripts) folder that can update all the `uses: Lombiq/GitHub-Actions/...@dev` refs to `uses: Lombiq/GitHub-Actions/...@issue/xxx-123`. Then the management issue is solved a single PowerShell script the developer can run manually before they push their pull request, or after they get the warnings from the failed workflow check. 

What do you think? As someone who reviews PRs and then has to merge them, does adding that `Update-GHA-Refs-to-Branch.ps1` script make more sense?

Reply:
The scripts sounds good to me!